float64 x
float64 y
---
float64 angle1
float64 angle2